### PR TITLE
Better telemetry for 409 errors

### PR DESCRIPTION
**Pull request by vladsud**
Has weak dependency on https://github.com/microsoft/FluidFramework/pull/7363

### CODE REVIEW

File: packages/drivers/odsp-driver/src/retryUtils.ts [hunk 1]
```diff
@@ -54,6 +69,7 @@ export async function runWithRetry<T>(
             assert(canRetry, "can retry");
             await delay(Math.floor(retryAfter));
             retryAfter += retryAfter / 4  * (1 + Math.random());
```
Comment (andre4i) [inline packages/drivers/odsp-driver/src/retryUtils.ts:71]: Unrelated to the change, I recall Aryn wrote a pretty neat throttler with both exponential and linear retry delays. Would that be useful in this context?

File: packages/drivers/odsp-driver/src/retryUtils.ts [hunk 2]
```diff
@@ -20,12 +20,24 @@ export async function runWithRetry<T>(
...
         try {
-            return await api();
+            const result = await api();
+            if (attempts > 1) {
+                logger.sendTelemetryEvent(
+                    {
+                        eventName: "MultipleRetries",
+                        callName,
+                        attempts,
+                        duration: performance.now() - start,
```
Comment (andre4i) [inline packages/drivers/odsp-driver/src/retryUtils.ts:36]: Just to make sure this is intentional, we're interested in the total time it took for the operation to succeed, not for how long it took this particular attempt, right?

**Review by andre4i — APPROVED** [review]

File: packages/drivers/odsp-driver/src/retryUtils.ts (hunk 2 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/retryUtils.ts:36]: Yep, it is

File: packages/drivers/odsp-driver/src/retryUtils.ts (hunk 1 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/retryUtils.ts:71]: Maybe? Hard to say, I'd need first to understand how it works :) So maybe good exercise for separate issue/PR

**Pull request merged**